### PR TITLE
[scripts] vcpkg_configure_make: do not set --build on macOS

### LINKED ISSUE #16464: [nettle] fails to build an x64-osx triplet on Apple Silicon
**Host Environment**
- OS: macOS 11.2.2 (20D80) 
- Compiler: Apple clang version 12.0.0 (clang-1200.0.32.29)

**To Reproduce**
`./vcpkg install nettle`

**Failure logs** 
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: /bin/bash ./../src/48e146e54f-e59f545639.clean/configure --build=arm64-apple-darwin --host=x86_64-apple-darwin --disable-documentation --disable-openssl --disable-shared --disable-silent-rules --verbose --disable-shared --enable-static --prefix=/Users/cromo/Documents/Sourcecode/vcpkg/installed/x64-osx/debug --bindir=${prefix}/../tools/nettle/debug/bin --sbindir=${prefix}/../tools/nettle/debug/sbin --includedir=${prefix}/../include --datarootdir=${prefix}/share/nettle
```
`Sourcecode/vcpkg/buildtrees/nettle/x64-osx-dbg/config-x64-osx-dbg-err.log`:
```
Invalid configuration `arm64-apple-darwin': machine `arm64-apple' not recognized
configure: error: /bin/sh ./../src/48e146e54f-e59f545639.clean/./config.sub arm64-apple-darwin failed
```

**Additional context**
Notice that this is a x64 (Rosetta) build, but the `arm64-apple-darwin` triplet is being incorrectly passed to `configure`.

### PR DESCRIPTION
**Pull request by autoantwort**
**Describe the pull request**

- What does your PR fix? Fixes #16464

See last discussion on #16595

### CODE REVIEW

**Comment by Neumann-A** [comment]
closes #16595
closes #16464
wasn't there another PR of you which touched the `BUILD_TRIPLET`?

**Comment by autoantwort** [comment]
> wasn't there another PR of you which touched the `BUILD_TRIPLET`?

None of mine, or not in that way. But I have multiple PRs that fix apple silicon builds.

**Comment by autoantwort** [comment]
Hm, but this breaks for example `gmp`
[build-x64-osx-dbg-err.log](https://github.com/microsoft/vcpkg/files/6101638/build-x64-osx-dbg-err.log)

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 1]
```diff
@@ -437,12 +437,9 @@ function(vcpkg_configure_make)
             # --target: the machine that CC will produce binaries for
             # https://stackoverflow.com/questions/21990021/how-to-determine-host-value-for-configure-when-using-cross-compiler
             # Only for ports using autotools so we can assume that they follow the common conventions for build/target/host
-            set(_csc_BUILD_TRIPLET "--build=${BUILD_ARCH}-apple-darwin")
             if(NOT "${TARGET_ARCH}" STREQUAL "${BUILD_ARCH}") # we don't need to specify the additional flags if we build natively.
-
-                list(APPEND _csc_BUILD_TRIPLET "--host=${TARGET_ARCH}-apple-darwin") # (Host activates crosscompilation; The name given here is just the prefix of the host tools for the target)
+                list(APPEND "--host=${TARGET_ARCH}-apple-darwin") # (Host activates crosscompilation; The name given here is just the prefix of the host tools for the target)
```
Comment (strega-nil) [inline scripts/cmake/vcpkg_configure_make.cmake:441]: this is definitely not valid CMake; you need `list(APPEND <list> <elements)`, but you don't have a list to append to.

**Review by strega-nil — CHANGES_REQUESTED** [review]

**Comment by Neumann-A** [comment]
hmm gmp seems to be on skip on osx?

File: scripts/cmake/vcpkg_configure_make.cmake (hunk 1 above)
Comment (autoantwort) [reply scripts/cmake/vcpkg_configure_make.cmake]: Oops! Fixed

**Comment by autoantwort** [comment]
> hmm gmp seems to be on skip on osx?

now with the valid cmake script it builds successfully on my machine

**Comment by strega-nil** [comment]
@autoantwort if it works, can you remove the skip in ci.baseline.txt?

**Comment by autoantwort** [comment]
> @autoantwort if it works, can you remove the skip in ci.baseline.txt?

There is no skip on osx for `gmp` or `nettle`?

**Comment by PhoebeHui** [comment]
gmp with osx and linux skipped by 'supports' which will be fixed by PR https://github.com/microsoft/vcpkg/pull/15957.

I suggest to fix the issue in PR https://github.com/microsoft/vcpkg/pull/16478, and after the PR merged, we should rerun this PR.

**Comment by ras0219-msft** [comment]
#15957 has been merged

**Comment by autoantwort** [comment]
I can not reproduce the build errors. Maybe simply rerun the pipeline?

**Comment by PhoebeHui** [comment]
@autoantwort, I have rerun the CI testing.

**Comment by wrobelda** [comment]
The `libgnutls` error is unrelated to this PR. I also cannot build it locally from master, even though I could at the time when I created a PR for it (https://github.com/microsoft/vcpkg/pull/14242). It should be added as skip in ci.baseline.txt for the time being, I'll look into fixing it.

**Comment by PhoebeHui** [comment]
The libgnutls:x64-osx failed with following failures:
```
Undefined symbols for architecture x86_64:
  "_libintl_dgettext", referenced from:
      _print_one_paragraph in libopts.a(libopts_la-libopts.o)
  "_libintl_gettext", referenced from:
      _aoGetsText in libcmd-serv.a(serv-args.o)
      _print_one_paragraph in libopts.a(libopts_la-libopts.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [gnutls-serv] Error 1
ld: warning: directory not found for option '-L/Users/vagrant/Data/installed/x64-osx/debug/lib/manual-link'
ld: warning: directory not found for option '-L/Users/vagrant/Data/installed/x64-osx/debug/lib/manual-link/'
Undefined symbols for architecture x86_64:
  "_libintl_dgettext", referenced from:
      _print_one_paragraph in libopts.a(libopts_la-libopts.o)
  "_libintl_gettext", referenced from:
      _aoGetsText in libcmd-cli.a(cli-args.o)
      _print_one_paragraph in libopts.a(libopts_la-libopts.o)
ld: warning: directory not found for option '-L/Users/vagrant/Data/installed/x64-osx/debug/lib/manual-link'
ld: warning: directory not found for option '-L/Users/vagrant/Data/installed/x64-osx/debug/lib/manual-link/'
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [gnutls-cli] Error 1
Undefined symbols for architecture x86_64:
  "_libintl_dgettext", referenced from:
      _print_one_paragraph in libopts.a(libopts_la-libopts.o)
  "_libintl_gettext", referenced from:
      _print_one_paragraph in libopts.a(libopts_la-libopts.o)
      _aoGetsText in libcmd-srp.a(srptool-args.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [srptool] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

```

**Comment by wrobelda** [comment]
@PhoebeHui yes, this is the issue I was referring to.

**Comment by PhoebeHui** [comment]
@wrobelda, thanks for confirming this issue!

**Comment by autoantwort** [comment]
I think `x264` will still fail, but on my arm mac it even fails before this change. Did someone have an intel mac and can look why it fails with this change?

**Comment by wrobelda** [comment]
Can confirm this fixes #16464 for me (nettle x64 build on Apple Silicon)

**Comment by ras0219-msft** [comment]
This LGTM, thanks everyone!

**Pull request merged**